libgnucash/engine/gnc-commodity.cpp

```diff
@@ -318,10 +318,13 @@ gnc_quote_source_fq_version (void)
 static QuoteSourceVec&
 get_quote_source_from_type (QuoteSourceType type)
 {
-    for (const auto& [s_type, sources] : quote_sources_map)
-        if (type == s_type)
-            return sources;
+    auto quote_sources_it = std::find_if (quote_sources_map.begin(), quote_sources_map.end(),
```

Comment:
I also notice you replaced my use of `unknown_as_default` construct. What was wrong with it ? Too exotic or is it because you wanted to keep the log warning in case an invalid quote source was used ?

Reply:
The latter. Also this branch includes slight change of behaviour when the quote source type is invalid: some functions return `new_quote_sources` info; others return 0. I'm pondering what's the best approach to mimic previous behaviour.

Reply:
Ok, that's reasonable. I didn't compare to how it worked originally. I have just followed your changes. So feel free to tweak to mimic previous behaviour.

Reply:
I rechecked: all functions with an invalid QuoteSourceType will pick `new_quote_sources` instead.